src/Features/CSharp/Portable/Braces/CSharpBracesDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,404 @@
+                        usingStatement.UsingKeyword.GetLocation(), SyntaxFacts.GetText(SyntaxKind.UsingKeyword));
+                }
+            }
+
+            if (node.IsKind(SyntaxKind.LockStatement))
+            {
+                var lockStatement = (LockStatementSyntax)node;
+                if (AnalyzeBlockedLockStatement(lockStatement))
+                {
+                    return Diagnostic.Create(_severityToRemoveBracesDescriptorMap[severity],
+                        lockStatement.LockKeyword.GetLocation(), SyntaxFacts.GetText(SyntaxKind.LockKeyword));
+                }
+            }
+
+            if (node.IsKind(SyntaxKind.FixedStatement))
+            {
+                var fixedStatement = (FixedStatementSyntax)node;
+                if (AnalyzeBlockedFixedStatement(fixedStatement))
+                {
+                    return Diagnostic.Create(_severityToRemoveBracesDescriptorMap[severity],
+                        fixedStatement.FixedKeyword.GetLocation(), SyntaxFacts.GetText(SyntaxKind.LockKeyword));
+                }
+            }
+
+            return null;
+        }
+
+        private bool AnalyzeIfStatement(IfStatementSyntax ifStatement) =>
+            !ifStatement.Statement.IsKind(SyntaxKind.Block);
+
+        private bool AnalyzeElseClause(ElseClauseSyntax elseClause) =>
+            !elseClause.Statement.IsKind(SyntaxKind.Block) &&
+            !elseClause.Statement.IsKind(SyntaxKind.IfStatement);
+
+        private bool AnalyzeForStatement(ForStatementSyntax forStatement) =>
+            !forStatement.Statement.IsKind(SyntaxKind.Block);
+
+        private bool AnalyzeForEachStatement(CommonForEachStatementSyntax forEachStatement) =>
+            !forEachStatement.Statement.IsKind(SyntaxKind.Block);
+
+        private bool AnalyzeWhileStatement(WhileStatementSyntax whileStatement) =>
+            !whileStatement.Statement.IsKind(SyntaxKind.Block);
+
+        private bool AnalyzeDoStatement(DoStatementSyntax doStatement) =>
+            !doStatement.Statement.IsKind(SyntaxKind.Block);
+
+        private bool AnalyzeUsingStatement(UsingStatementSyntax usingStatement) =>
+            !usingStatement.Statement.IsKind(SyntaxKind.Block) &&
+            !usingStatement.Statement.IsKind(SyntaxKind.UsingStatement);
+
+        private bool AnalyzeLockStatement(LockStatementSyntax lockStatement) =>
+            !lockStatement.Statement.IsKind(SyntaxKind.Block) &&
+            !lockStatement.Statement.IsKind(SyntaxKind.LockStatement);
+
+        private bool AnalyzeFixedStatement(FixedStatementSyntax fixedStatement) =>
+            !fixedStatement.Statement.IsKind(SyntaxKind.Block) &&
+            !fixedStatement.Statement.IsKind(SyntaxKind.FixedStatement);
+
+        private bool AnalyzeBlockedIfStatement(IfStatementSyntax ifStatement) =>
+            ifStatement.Statement.IsKind(SyntaxKind.Block) &&
+            ((BlockSyntax)ifStatement.Statement).Statements.Count == 1;
```

Comment:
this pattern is repeated everywhere below.   can you extact out common code.  like "IsBlockWithSingleStatement(stmt)"